atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraServersConfigs.java

```diff
@@ -149,4 +155,24 @@ public boolean validateHosts() {
             return visitor.visit(this);
         }
     }
+
+    public static <T> Optional<T> deriveFromCqlHosts(
```

Comment:
Maybe this should just return an `Optional<CqlCapableConfig>` and then you can map it as you wish? This is a little bit strange